pkg/util/netutil/net.go

```diff
@@ -150,12 +153,15 @@ func (s *Server) ServeWith(
 			return e
 		}
 		tempDelay = 0
-		go func() {
+		err := stopper.RunAsyncTask(ctx, "pgwire-serve", func(ctx context.Context) {
 			defer stopper.Recover(ctx)
 			s.Server.ConnState(rw, http.StateNew) // before Serve can return
-			serveConn(rw)
+			serveConn(ctx, rw)
 			s.Server.ConnState(rw, http.StateClosed)
```

Comment:
might be worth `defer` on the 2nd ConnState call here, since youre in this area.